homeassistant/components/inels/config_flow.py

```diff
@@ -0,0 +1,180 @@
+"""Config flow for iNELS."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from inelsmqtt import InelsMqtt
+from inelsmqtt.const import MQTT_TRANSPORT
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN, TITLE
+
+CONNECTION_TIMEOUT = 5
+
+
+class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle of iNELS config flow."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> InelsOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return InelsOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        return await self.async_step_setup()
+
+    async def async_step_setup(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the setup step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            test_connect = await self.hass.async_add_executor_job(
+                try_connection,
+                self.hass,
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+                user_input.get(CONF_USERNAME, ""),
+                user_input.get(CONF_PASSWORD, ""),
+                user_input[MQTT_TRANSPORT],
+            )
+
+            if test_connect is None:
+                return self.async_create_entry(
+                    title=TITLE,
+                    data=user_input,
+                )
+
+            errors["base"] = connect_val_to_error(test_connect)
+
+        return self.async_show_form(
+            step_id="setup",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_PORT, default=1883): vol.Coerce(int),
+                    vol.Optional(CONF_USERNAME): str,
+                    vol.Optional(CONF_PASSWORD): str,
+                    vol.Required(MQTT_TRANSPORT, default="tcp"): vol.In(
+                        ["tcp", "websockets"]
+                    ),
```

Comment:
Why do we ask the user for the transport method?

Reply:
You're right that TCP is the default and most common option. However, the Paho MQTT library does support both TCP and WebSockets, which could be useful in certain network configurations.

Reply:
I include the transport method option because in certain network scenarios, especially when the MQTT broker is not running as an add-on, WebSockets may solve connectivity issues.